components/EditEvent.js

```diff
@@ -41,6 +47,23 @@ class EditEvent extends React.Component {
     }
   }
 
+  async deleteEvent() {
+    this.setState({
+      deleteLoading: true,
+    });
+    try {
+      const { event } = this.props;
+      await this.props.deleteCollective(event.id);
+      this.setState({ result: { success: 'Event deleted successfully' } });
+      Router.pushRoute(`/${event.parentCollective.slug}`);
+    } catch (err) {
+      console.error('>>> deleteEvent error: ', JSON.stringify(err));
+      const errorMsg = err.graphQLErrors && err.graphQLErrors[0] ? err.graphQLErrors[0].message : err.message;
+      this.setState({ result: { error: errorMsg } });
+      throw new Error(errorMsg);
```

Comment:
Why re-throwing here?

Reply:
I followed the same convention which was with `editEvent` - https://github.com/opencollective/opencollective-frontend/pull/2803/files/b92147d15867a26b29f94242b4144ee97a57d380#diff-5b1cedbc22f5352cc2104058e2854264R46

Reply:
If the error is already handled, this doesn't seems necessary

Reply:
yeah! should I also remove it from `editEvent` handler?

Reply:
I didn't check the implications for `editEvent`. It may be out of scope, but if you think that's safe you can go ahead and make the change.